tar/read.c

```diff
@@ -265,6 +299,20 @@ read_archive(struct bsdtar *bsdtar, char mode)
 			if (edit_pathname(bsdtar, entry))
 				continue; /* Excluded by a rewrite failure. */
 
+			/* Don't extract if the file already matches it. */
+			if (bsdtar->option_resume_extract) {
+				if ((r = check_skip_file(
+				    archive_entry_pathname(entry), st)) == -1) {
```

Comment:
How carefully did you check that `archive_entry_pathname(entry)` is the correct path to the file we would create?  I *think* this is correct but with options like `-C` and `-s` and the handling of absolute paths and paths containing `..` I want to be sure that we have this right and aren't accidentally comparing metadata against the wrong place.